app/v2/models/MessageType.scala

```diff
@@ -25,25 +25,31 @@ import play.api.libs.json.Writes
 
 object MessageType {
-  case object AmendmentAcceptance              extends MessageType("IE004", "CC004C")
-  case object InvalidationDecision             extends MessageType("IE009", "CC009C")
-  case object PositiveAcknowledgement          extends MessageType("IE928", "CC928C")
-  case object MRNAllocated                     extends MessageType("IE028", "CC028C")
-  case object GoodsReleaseNotification         extends MessageType("IE025", "CC025C")
-  case object ReleaseForTransit                extends MessageType("IE029", "CC029C")
-  case object UnloadingPermission              extends MessageType("IE043", "CC043C")
-  case object RejectionFromOfficeOfDestination extends MessageType("IE057", "CC057C")
-  case object RequestOnNonArrivedMovementDate  extends MessageType("IE140", "CC140C")
-  case object RejectionFromOfficeOfDeparture   extends MessageType("IE056", "CC056C")
-  case object ControlDecisionNotification      extends MessageType("IE060", "CC060C")
-  case object RecoveryNotification             extends MessageType("IE035", "CC035C")
-  case object NoReleaseForTransit              extends MessageType("IE051", "CC051C")
-  case object GuaranteeNotValid                extends MessageType("IE055", "CC055C")
+  case object AmendmentAcceptance               extends MessageType("IE004", "CC004C")
+  case object InvalidationDecision              extends MessageType("IE009", "CC009C")
+  case object PositiveAcknowledgement           extends MessageType("IE928", "CC928C")
+  case object MRNAllocated                      extends MessageType("IE028", "CC028C")
+  case object GoodsReleaseNotification          extends MessageType("IE025", "CC025C")
+  case object ReleaseForTransit                 extends MessageType("IE029", "CC029C")
+  case object UnloadingPermission               extends MessageType("IE043", "CC043C")
+  case object RejectionFromOfficeOfDestination  extends MessageType("IE057", "CC057C")
+  case object RequestOnNonArrivedMovementDate   extends MessageType("IE140", "CC140C")
+  case object RejectionFromOfficeOfDeparture    extends MessageType("IE056", "CC056C")
+  case object ControlDecisionNotification       extends MessageType("IE060", "CC060C")
+  case object RecoveryNotification              extends MessageType("IE035", "CC035C")
+  case object NoReleaseForTransit               extends MessageType("IE051", "CC051C")
+  case object GuaranteeNotValid                 extends MessageType("IE055", "CC055C")
+  case object Discrepancies                     extends MessageType("IE019", "CC019C")
+  case object WriteOffNotification              extends MessageType("IE045", "CC045C")
+  case object ForwardedIncidentNotificationToED extends MessageType("IE182", "CC182C")
 
   val arrivalMessages: Seq[MessageType] = Seq(
     GoodsReleaseNotification,
     UnloadingPermission,
-    RejectionFromOfficeOfDestination
+    RejectionFromOfficeOfDestination,
+    Discrepancies,
+    WriteOffNotification,
```

Comment:
This is a departure message -- rather, it's sent by the office of departure to a trader's departure.

Reply:
it's defined as Arrival message here - https://docs.google.com/spreadsheets/d/17rXkMJqB9uLWAEdV2v9ob2MiqAbPidfX4zMtiv5ieOg/edit#gid=391554264

and also in the ticket - https://jira.tools.tax.service.gov.uk/browse/CTCP-2083

Reply:
That's not quite right and we need to get that updated, I'll talk to the right people about that. The IE045 message is sent to the departure trader, to indicate that an arrival has finished. So in the sense of this, IE045 is a **departure** message.

Reply:
as per this spec https://drive.google.com/file/d/1QVYsezQUfVm6ajeC6Kp9sG5821HKBQ7g/view
CC019C --> Arrival
CC045C --> Arrival
CC182C --> Departure

Maybe this need to be updated as well

Reply:
That has been updated since, that's an old version!